build_tools/bazel_to_cmake/bazel_to_cmake_converter.py

```diff
@@ -27,207 +27,151 @@
+  return _convert_sorted_string_list_block(list_name, targets)
+
+
 class BuildFileFunctions(object):
   """Object passed to `exec` that has handlers for BUILD file functions."""
 
   def __init__(self, converter):
     self.converter = converter
 
-  # ------------------------------------------------------------------------- #
-  # Conversion utilities, written to reduce boilerplate and allow for reuse   #
-  # between similar rule conversions (e.g. cc_library and cc_binary).         #
-  # ------------------------------------------------------------------------- #
-
-  def _convert_string_arg_block(self, name, value):
-    #  NAME
-    #    value
-    return f"  {name}\n    {value}\n"
-
-  def _convert_name_block(self, name):
-    #  NAME
-    #    rule_name
-    return f"  NAME\n    {name}\n"
-
-  def _convert_out_block(self, out):
-    #  OUT
-    #    out_name
-    return f"  OUT\n    {out}\n"
-
-  def _convert_cc_namespace_block(self, cc_namespace):
-    #  CC_NAMESPACE
-    #    "cc_namespace"
-    if not cc_namespace:
-      return ""
-    return f'  CC_NAMESPACE\n    "{cc_namespace}"\n'
-
-  def _convert_cpp_namespace_block(self, cpp_namespace):
-    #  CPP_NAMESPACE
-    #    "cpp_namespace"
-    if not cpp_namespace:
-      return ""
-    return f'  CPP_NAMESPACE\n    "{cpp_namespace}"\n'
-
-  def _convert_string_list_block(self, name, values):
-    # Note this deliberately distinguishes between an empty list (argument
-    # explicitly specified) and None (argument left as default).
-    if values is None:
-      return ""
-    values_list = "\n".join([f'    "{v}"' for v in values])
-    return f"  {name}\n{values_list}\n"
-
-  def _convert_translate_tool_block(self, translate_tool):
-    if translate_tool and translate_tool != "//iree/tools:iree-translate":
-      # Bazel `//iree/base`     -> CMake `iree::base`
-      # Bazel `//iree/base:api` -> CMake `iree::base::api`
-      translate_tool = translate_tool.replace("//iree", "iree")  # iree/base:api
-      translate_tool = translate_tool.replace(":", "_")  # iree/base::api
-      translate_tool = translate_tool.replace("/", "_")  # iree::base::api
-      return f"  TRANSLATE_TOOL\n    {translate_tool}\n"
-    else:
-      return ""
-
-  def _convert_option_block(self, option, option_value):
-    if option_value:
-      # Note: this is a truthiness check as well as an existence check, i.e.
-      # Bazel `testonly = False` will be handled correctly by this condition.
-      return f"  {option}\n"
-    else:
-      return ""
-
-  def _convert_testonly_block(self, testonly):
-    return self._convert_option_block("TESTONLY", testonly)
-
-  def _convert_flatten_block(self, flatten):
-    return self._convert_option_block("FLATTEN", flatten)
-
-  def _convert_file_list_block(self, list_name, files):
-    #  list_name
-    #    "file_1.h"
-    #    "file_2.h"
-    #    "file_3.h"
-    if not files:
-      return ""
-    files_list = "\n".join([f'    "{file}"' for file in sorted(files)])
```

Comment:
IIRC, the quoting for file names was first introduced in the hand-ported CMake files and then I copied that style over to the scripts. I'm not sure if there's value in continuing that styling.

If you proceed with removing the quotes around file names, can you also update the documentation to match?

https://github.com/google/iree/blob/cf1047322db8e591c25cc27de954d8d68173d9e4/build_tools/cmake/iree_cc_library.cmake#L42-L66

https://github.com/google/iree/blob/cf1047322db8e591c25cc27de954d8d68173d9e4/build_tools/cmake/iree_cc_binary.cmake#L35-L55

https://github.com/google/iree/blob/cf1047322db8e591c25cc27de954d8d68173d9e4/build_tools/cmake/external_cc_library.cmake#L46-L78

etc.

---

OR you could keep the style changes out of this commit/PR and make them in isolation.

Reply:
Yeah I didn't intend to do the style change as part of this, but while trying to unify things discovered what appeared to be arbitrary quoting distinctions between different types of arguments. In classic CMake style, searching around reveals that there are no official recommendations on when to quote things and the unofficial recommendations from people are contradictory. So... I just picked something and it seems to all build fine without quotes. If someone with past CMake experience has a strong opinion on when things should be quoted, I'm happy to implement that. I'll save the manual updates until we decide where this is going though. I could also start out with separate quoted/unquoted variations for each string function and preserve the existing style

Reply:
I added optional quoting to the helper functions and restored the previous quoting behavior. We can continue to discuss what the right thing to do here is.

Reply:
SG. I'm not against the quoting changes, but I'd prefer for them to be made in a separate commit that includes the documentation updates.